### PR TITLE
Added edit mode to data center view

### PR DESCRIPTION
**Pull request by ar4s**
edit

Changes:
- added modes - ``preview`` and ``edit``
modes

- preview mode:
 - no changes,
- edit mode:
 - each rack can be move by drag and drop method
green

 - drag and drop validation
validation

 - each rack can be rotate
rotate

 - name and description of each rack can be edited
edit

### CODE REVIEW

**Comment by vi4m** [comment]
Work in progress still ? What is missing ? :-)

**Comment by vi4m** [comment]
Apart from good description obviously

File: src/ralph/ui/static/partials/data_center/rack.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<div class="rack rotate-{{ rack.orientation }} x-{{ rack.visualization_col }} y-{{ rack.visualization_row }}">
-    <a href="#/rack/{{ rack.id }}" class="wrapper">
+<div class="rack rotate-{{ rack.orientation }} x-{{ rack.visualization_col }} y-{{ rack.visualization_row }}" ng-class="{'move': is_move,'allowed': can_drop, 'not_allowed': !can_drop}">
```
Comment (vi4m) [inline src/ralph/ui/static/partials/data_center/rack.html]: ad space before  , here

File: src/ralph/ui/static/ui/js/app/data_center/directives.js [hunk 2]
```diff
@@ -2,12 +2,65 @@
...
+                    $document.unbind('mousemove', mousemove);
+                    $document.unbind('mouseup', mouseup);
+                    scope.rack.$update_rack({dcId: scope.datacenter.id});
+                    scope.is_move = false;
+                }
+
+                scope.rotate = function(direction) {
+                    var orientations = ['top', 'right', 'bottom', 'left'];
+                    var index = orientations.indexOf(scope.rack.orientation) + direction;
+                    if (index < 0)
```
Comment (vi4m) [inline src/ralph/ui/static/ui/js/app/data_center/directives.js]: always use { in if

**Comment by vi4m** [comment]
1. We can't merge it 
2. Squash it baby

**Comment by quamilek** [comment]
Would do well to explain, put short message under the header  as to what it really can be edited and how.

**Comment by xliiv** [comment]
:ok:

**Comment by andrzej-jankowski** [comment]
:ok:

**Pull request merged**